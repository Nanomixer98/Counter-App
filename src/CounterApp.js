import React, { useState } from 'react';
import PropTypes from 'prop-types';

const CounterApp = ({ value = 3000 }) => {

    const [ counter, setCounter ] = useState(value); // []

    // handleAdd
    const handleAdd = (e) => {
        setCounter(counter + 1);
    }

    // handleSub
    const handleSubtract = (e) => {
        // setCounter(counter + 1);
        setCounter( (c) => c - 1 )
    }

    return (<>
        <h1>Wellcome to first CounterApp</h1>
        <h2>{ counter }</h2>

        <button onClick={ handleAdd }>+1</button>
        <button onClick={ () => setCounter( value ) }>Reset</button>
        <button onClick={ handleSubtract }>-1</button>
    </>);
}

CounterApp.prototypes = {
    value: PropTypes.number
}

export default CounterApp;