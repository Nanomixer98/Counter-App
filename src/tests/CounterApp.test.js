import CounterApp from "../CounterApp";
import "@testing-library/jest-dom";
import React from 'react';
import { shallow } from 'enzyme';

let wrapper = shallow(<CounterApp/>);
beforeEach(() => {
    wrapper = shallow(<CounterApp />);
})

describe('Pruebas en el componente CounterApp', () => {

  test('Mostrar correctamente <CounterApp /> con valores por defecto', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Mostrar correctamente <CounterApp /> con valor de 100', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value}/>);

        const valueRendered = wrapper.find('h2').text().trim();
        expect(parseInt(valueRendered)).toBe(value);
  });

});

describe('Funcionamiento de botones', () => {
  
    test('Debe incrementar en +1', () => {
        // const wrapper = shallow(<CounterApp/>);
        wrapper.find('button').at(0).simulate('click');

        const valueRendered = wrapper.find('h2').text().trim();
        expect(parseInt(valueRendered)).toBe(3001);
    });
  
    test('Debe disminuir -1', () => {
        // const wrapper = shallow(<CounterApp/>);
        wrapper.find('button').at(2).simulate('click');

        const valueRendered = wrapper.find('h2').text().trim();
        expect(parseInt(valueRendered)).toBe(2999);
    });
  
    test('Debe de regresar el valor a defecto "100"', () => {
        const value = 100;
        const wrapper = shallow(<CounterApp value={value}/>);
        // const wrapper = shallow(<CounterApp/>);
        for (let index = 0; index < 8; index++) {
            wrapper.find('button').at(2).simulate('click');
        }
        for (let index = 0; index < 3; index++) {
            wrapper.find('button').at(0).simulate('click');
        }
        wrapper.find('button').at(1).simulate('click');

        const valueRendered = wrapper.find('h2').text().trim();
        expect(parseInt(valueRendered)).toBe(100);
    });

});

