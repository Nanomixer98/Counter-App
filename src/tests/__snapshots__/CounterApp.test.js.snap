// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente CounterApp Mostrar correctamente <CounterApp /> con valores por defecto 1`] = `
<Fragment>
  <h1>
    Wellcome to first CounterApp
  </h1>
  <h2>
    3000
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
